sycl/doc/extensions/experimental/sycl_ext_oneapi_virtual_mem.asciidoc

```diff
@@ -0,0 +1,363 @@
+throws a synchronous exception with the `errc::invalid` error code.
+
+This will allocate `numBytes` of physical memory on the device. `numBytes` must
+be a multiple of the granularity, as returned by a call to
+`get_mem_granularity`.
+
+If `syclDevice` does not have `aspect::ext_oneapi_virtual_mem` the call throws
+an exception with `errc::feature_not_supported`.
+
+|`physical_mem(const queue &syclQueue, size_t numBytes)` |
+Same as `physical_mem(syclQueue.get_device(), syclQueue.get_context, numBytes)`.
+
+|`void *map(uintptr_t ptr, size_t numBytes, size_t offset = 0)` |
+Same as `map(ptr, numBytes, address_access_mode::none, offset)`.
+
+|`void *map(uintptr_t ptr, size_t numBytes, address_access_mode mode, size_t offset = 0)` |
+Maps a virtual memory range, specified by `ptr` and `numBytes`, to the physical
+memory corresponding to this instance of `physical_mem`, starting at an offset
+of `offset` bytes.
+
+If `mode` is `address_access_mode::read` or `address_access_mode::read_write`
+the returned pointer is accessible after the call as read-only or read-write
+respectively. Otherwise, it is considered inaccessible and accessing it will
+result in undefined behavior.
+
+The returned pointer is equivalent to `reinterpret_cast<void *>(ptr)`.
+
+Writing to any address in the virtual memory range with access mode set to
+`access_mode::read` results in undefined behavior.
+
+An accessible pointer behaves the same as a pointer to device USM memory and can
+be used in place of a device USM pointer in any interface accepting one.
+
+|`context get_context() const` |
+Returns the SYCL context associated with the instance of `physical_mem`.
+
+|`device get_device() const` |
+Returns the SYCL device associated with the instance of `physical_mem`.
+
+|`size_t size() const` |
+Returns the size of the corresponding physical memory in bytes.
+
+|============================
+
+Virtual memory address ranges are mapped to the a `physical_mem` through the
+`map` member functions, where the access mode can also be specified.
+To further get or set the access mode of a mapped virtual address range, the
+user does not need to know the associated `physical_mem` and can just call the
+following free functions.
+
+```c++
+namespace sycl::ext::oneapi::experimental {
+
+void set_access_mode(const void *ptr, size_t numBytes, address_access_mode mode, const context &syclContext);
+
+address_access_mode get_access_mode(const void *ptr, size_t numBytes, const context &syclContext);
+
+void unmap(const void *ptr, size_t numBytes, const context &syclContext);
+
+} // namespace sycl::ext::oneapi::experimental
+```
+
+[frame="topbot",options="header,footer"]
+|=====================
+|Function |Description
+
+|`void set_access_mode(const void *ptr, size_t numBytes, address_access_mode mode, const context &syclContext)` |
+Sets the access mode of a virtual memory range specified by `ptr` and
+`numBytes`.
```

Comment:
Must `ptr` and `numBytes` be the exact range from some previous call to `physical_mem::map`, or may they be a subrange?

If a subrange is allowed, do `ptr` and `numBytes` need to be a multiple of the minimum granularity?

Reply:
It seems like both backends allow sub-range for this, but L0 talks about "page size" but doesn't refer to the [zeVirtualMemQueryPageSize](https://spec.oneapi.io/level-zero/latest/core/api.html#zevirtualmemquerypagesize) query like other APIs, so I am unsure whether it refers to the host page size or the result of that query. I have assumed the latter.

Reply:
No, for L0 this would correspond to the LCM of `zeVirtualMemQueryPageSize` of all devices in that context. For CUDA it would be the host page size.

Reply:
I see that you changed the wording in 8dd8cb3b60ec8d8d4a2882fd326e5017e58f7dd5 to say that the `ptr` must be aligned to the device's minimum granularity.  This clarifies my concern about the wording, so I think we can resolve this issue.